specutils/utils/spectral_region.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+import astropy.units as u
+
+
+class SpectralRegion:
+    """
+    SpectralRegion is currently a container class enables some simplicty
+    in defining and passing a region (interval) for a spectrum.
+
+    In the future, there might be more functionality added in here and there
+    is some discussion that this might/could move to astropy.region.
+    """
+
+    def __init__(self, lower, upper):
+        """
```

Comment:
The usual convention we follow in astropy is to put the initializer arguments (i.e., the paramters section) in the *class* docstring.  That makes it more visible to the user because the average user doesn't know about (not care) that ``__init__`` is where the parameters are defined.  Also the documentation machinery doesn't recognize it if it's in the ``__init__`` docstring. (I think?)